aeron-driver/src/main/java/io/aeron/driver/PublicationImage.java

```diff
@@ -690,8 +695,14 @@ int sendPendingStatusMessage(final long nowNs)
     {
         int workCount = 0;
         final long changeNumber = endSmChange;
+        final boolean hasSmTimedOut = (timeOfLastSmNs + smTimeoutNs) - nowNs < 0;
 
-        if (changeNumber != lastSmChangeNumber || (timeOfLastSmNs + smTimeoutNs) - nowNs < 0)
+        if (null != responseSessionId && hasSmTimedOut)
```

Comment:
Double read of the volatile field `responseSessionId` (inside of the `if`). Also the order of the checks can be reversed to avoid reading that field id the `hasSmTimedOut == false`.